ASM feature (#96)
autonity/docs.autonity.org: content/docs/concepts/architecture/_index.md

- Set oracle voters based on validators in the consensus committee and update the voter set as the consensus committee is re-selected at the end of an epoch.
- Manage oracle voting rounds, triggering the initiation of a new voting period at the end of a round.

#### Voter selection

Participation in the oracle protocol is a validator responsibility and validators in the consensus committee are automatically selected to vote on median price computation by a protocol-only function. As the last block of an epoch is finalized, this function is executed to determine the oracle voters for the following epoch.

Consensus committee membership is computed by the Autonity Protocol Contract; see [committee selection](/concepts/architecture/#committee-selection).

### ASM ACU Contract

> The contract implementing the Auton Currency Unit (ACU) element of the Auton Stability Mechanism. The contract computes the value of the ACU, an optimal currency basket of 7 free-floating fiat currencies. Value is computed for the basket currencies using [median price data](/concepts/architecture/#median-price-computation) from the Oracle Contract. The basket quantity corresponding to each symbol is set to give ACU maximum stability.

Reviewer: ```suggestion
The contract implementing the Auton Currency Unit (ACU) element of the Auton Stability Mechanism. It computes the value of the ACU, an optimal currency basket of 7 free-floating fiat currencies. Value is computed for the basket currencies using [median price data](/concepts/architecture/#median-price-computation) from the Oracle Contract. The basket quantity corresponding to each symbol is set to give ACU maximum stability.
```

> The contract provides primitives for computing the ACU value and managing the basket currency symbols and quantities (i.e. weighting). The contract stores [protocol parameters](/reference/protocol/) that specify the currency pairs for the basket, the quantities of those currencies in the basket, and the scale of precision for the ACU value. Per the Autonity Protocol Contract, ACU protocol parameters are initialised at network [genesis](/reference/genesis/).

Reviewer: ```suggestion
The contract provides primitives for computing the ACU value and managing the basket currency symbols and quantities (i.e. weighting). It stores [protocol parameters](/reference/protocol/) that specify the currency pairs for the basket, the quantities of those currencies in the basket, and the scale of precision for the ACU value. Per the Autonity Protocol Contract, ACU protocol parameters are initialised at network [genesis](/reference/genesis/).
```

The contract provides primitives for managing the available supply of Auton in an Autonity network. The contract stores the [protocol parameter](/reference/protocol/) setting the network's available Auton supply. Per the Autonity Protocol Contract, ACU protocol parameters are initialised at network [genesis](/reference/genesis/).

The contract function for returning the available supply of Auton for minting can be called by all participants.  Function calls to mint and burn Auton are restricted to invocation by the protocol Stabilization Contract.

All functions are documented in the Reference [Autonity Interfaces](/reference/api/): public API's under [Supply Control Contract Interface](/reference/api/asm/supplycontrol/), mint and burn under [Governance and Protocol-Only Reference](/reference/api/aut/op-prot/).

#### Auton supply control

The Autonity Supply Control Contract manages the Auton supply. The contract implements logic to:

- Mint and burn of Auton to take Auton in and out of circulation.

To learn more about the concept see [Auton Stability Mechanism (ASM)](/concepts/asm/).

### ASM Stabilization Contract

> The contract implementing the CDP-based stabilization mechanism for the Auton. Auton is borrowed against Collateral Token using a Collateralized Debt Position (CDP) mechanism. The Stabilization Contract manages CDP's throughout the lifecycle, from initial borrowing through repayment and liquidation scenarios. Collateral Token is deposited into a CDP to borrow Auton. Auton is brought in and out of circulation on an Autonity network as CDP's are opened and closed.

Reviewer: The plural of CDP should be CDPs:

> The list of open CDPs

...whereas "CDP's" should be reserved for possessive:

> The CDP's outstanding debt

Further to this:

```suggestion
The contract implementing the CDP-based stabilization mechanism for the Auton. Auton is borrowed against Collateral Token using a Collateralized Debt Position (CDP) mechanism. The Stabilization Contract manages CDPs throughout the lifecycle, from initial borrowing through repayment and liquidation scenarios. Collateral Token is deposited into a CDP to borrow Auton. Auton is brought in and out of circulation on an Autonity network as CDPs are opened and closed.
```

Author: OK. That makes sense!

The contract provides primitives for stabilization configuration, CDP calculations, and CDP lifecycle management. The contract stores [protocol parameter](/reference/protocol/) setting the configuration of the stabilisation mechanism’s Collateralised Debt Position (CDP). Per the Autonity Protocol Contract, ACU protocol parameters are initialised at network [genesis](/reference/genesis/).

> Contract functions can be called by all participants to:

Reviewer: ```suggestion
Contract functions can be called by:

- By CDP owners to take out CDP's to borrow Auton, withdraw collateral, and repay CDP's.
- By prospective CDP owners to determine borrowing limits  and collateral level requirements

> - By CDP liquidators to determine if a CDP is liquidatable or not, and to liquidate CDP's.

Reviewer: ```suggestion
- By CDP owners to take out CDPs to borrow Auton, withdraw collateral, and repay CDPs.
- By prospective CDP owners to determine borrowing limits  and collateral level requirements
- By CDP liquidators to determine if a CDP is liquidatable or not, and to liquidate CDPs.
```
